ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/connections/netty/NettyTLSConnectionInitializer.java

```diff
@@ -69,7 +70,12 @@ public NettyTLSConnectionInitializer(
       final MetricsSystem metricsSystem,
       final Supplier<TLSContextFactory> tlsContextFactorySupplier) {
     super(nodeKey, config, localNode, eventDispatcher, metricsSystem);
-    this.tlsContextFactorySupplier = Optional.ofNullable(tlsContextFactorySupplier);
+    if (tlsContextFactorySupplier != null) {
+      this.tlsContextFactorySupplier =
+          Optional.of(Suppliers.memoize(tlsContextFactorySupplier::get));
+    } else {
+      this.tlsContextFactorySupplier = Optional.empty();
+    }
```

Comment:
The `memoize` was introduced because TLS Context was initialized multiple time which cause an issue with the SunPKCS11 + nss as nss only allows one connection to be established at a time. I'll double check if memoize returns null or not, if it does, it would become simpler.

Reply:
memoize doesn't work on nulls ... hence we need the non-null check.